pkg/cmd/cli/uninstall/uninstall.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2020 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package uninstall
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/vmware-tanzu/velero/pkg/client"
+	"github.com/vmware-tanzu/velero/pkg/cmd"
+	"github.com/vmware-tanzu/velero/pkg/cmd/util/output"
+	"github.com/vmware-tanzu/velero/pkg/install"
+)
+
+func Uninstall(c *cobra.Command, f client.Factory) error {
+	resources := install.AllCRDs()
+
+	dynamicClient, err := f.DynamicClient()
+	if err != nil {
+		return err
+	}
+
+	factory := client.NewDynamicFactory(dynamicClient)
+
+	veleroNs := strings.TrimSpace(f.Namespace())
+
+	if veleroNs == "" {
+		veleroNs = "velero"
+	}
+
+	ds := install.DaemonSet(veleroNs, []install.PodTemplateOption{}...)
+	if err := install.AppendUnstructured(resources, ds); err != nil {
+		return err
+	}
+
+	de := install.Deployment(veleroNs, []install.PodTemplateOption{}...)
+	if err := install.AppendUnstructured(resources, de); err != nil {
+		return err
+	}
+
+	se := install.Secret(veleroNs, nil)
+	if err := install.AppendUnstructured(resources, se); err != nil {
+		return err
+	}
+
+	sa := install.ServiceAccount(veleroNs, map[string]string{})
+	if err := install.AppendUnstructured(resources, sa); err != nil {
+		return err
+	}
+
+	crb := install.ClusterRoleBinding(veleroNs)
```

Comment:
Everything above this line is unnecessary, as the namespaced resources will be deleted with the namespace.

Also, we need to add the CustomResourceDefinitions. You can select them similarly to [this](https://github.com/vmware-tanzu/velero/blob/main/pkg/cmd/cli/plugin/helpers.go#L33)

Reply:
@nrb , makes sense. I am using an existing helper function to get all the CRDs 
https://github.com/vmware-tanzu/velero/blob/827bad0ecff0f00d80fdd39c1d06e2ff5722c8b7/pkg/cmd/cli/uninstall/uninstall.go#L36

Do you think using labels would be a better approach?

Reply:
Yeah, the `AllCRDs` function was really meant to be used on installation only, not necessarily re-used as a selector.

Like you said, version skew could make this inaccurate.

Reply:
Yeah. Will make the change then.

Reply:
Please take a look at the VeleroUninstall in https://github.com/vmware-tanzu/velero/blob/main/test/e2e/velero_utils.go

We can expand on this as necessary, but it is pretty much a straight port of the existing hand uninstall process.  
When "velero uninstall" is available in the CLI, the plan is to call that from the test code.

Reply:
@dsu-igeek , I think we can port the uninstall code directly for the uninstall command. 

>  When "velero uninstall" is available in the CLI, the plan is to call that from the test code.
Is this so that we could test the cli or is it okay to directly call the uninstall function? Asking this because I see we are doing the latter for `VeleroInstall` in `test/e2e/velero_utils.go`

Reply:
We should eventually call the CLI both for install and uninstall.  That way we can test the CLI on those paths as part of the testing.